capstone-backend/src/main/java/com/google/sps/data/BookList.java

```diff
@@ -17,77 +17,89 @@
  */
 public final class BookList {
 
-  private Long id;
-  private Long userID;
-  private String bookshelfID;
-  private Collection<Long> collaboratorsIDs;
+  private String id;
+  private String userID;
+  private Collection<String> gbookIDs;
+  private Collection<String> collaboratorsIDs;
 
-  public BookList(Long userID, String bookshelfID, Collection<Long> collaboratorsIDs, Long id) {
+  public BookList(String id, String userID, Collection<String> gbookIDs, Collection<String> collaboratorsIDs) {
+    this.id = id;
     this.userID = userID;
-    this.bookshelfID = bookshelfID;
+    this.gbookIDs = gbookIDs;
     this.collaboratorsIDs = collaboratorsIDs;
-    this.id = id;
   }
 
-  public BookList(Long userID, String bookshelfID, Collection<Long> collaboratorsIDs) {
-    this(userID, bookshelfID, collaboratorsIDs, null);
+  public BookList(String userID, Collection<String> gbookIDs, Collection<String> collaboratorsIDs) {
+    this("", userID, gbookIDs, collaboratorsIDs);
   }
 
-  public BookList(Long userID, String bookshelfID){
-    this(userID, bookshelfID, new HashSet<Long>(), null);
+  public BookList(String userID, Collection<String> gbookIDs) {
+    this("", userID, gbookIDs, new HashSet<String>());
   }
 
-  public void setBookshelf(String bookshelfID) {
-    this.bookshelfID = bookshelfID;
+  public void setGbookIDs(Collection<String> gbookIDs) {
+    this.gbookIDs = gbookIDs;
   }
 
-  public void setUserID(Long userID) {
+  public void setUserID(String userID) {
     this.userID = userID;
   }
 
-  public void setCollaboratorsIDs(Collection<Long> collaboratorsIDs) {
+  public void setCollaboratorsIDs(Collection<String> collaboratorsIDs) {
     this.collaboratorsIDs = collaboratorsIDs;
   }
 
-  public void setID(Long id){
+  public void setID(String id) {
     this.id = id;
   }
 
   public boolean isEmpty() {
-    return userID == null && bookshelfID.equals("") && collaboratorsIDs.isEmpty();
+    return userID.equals("") && gbookIDs.isEmpty() && collaboratorsIDs.isEmpty();
   }
 
   public void clear() {
```

Comment:
We might not need this. I don't think it would hurt, but we can delete entities in the firestore rather than clear all their fields here.

Reply:
Okay